src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -2672,10 +2770,24 @@ HRESULT CProfilerManager::InitializeForAttach(
 {
     HRESULT hr = S_OK;
 
-    // TODO: How to handle this? Does this need to support profiler attach? How does that impact initlization etc...
-
     PROF_CALLBACK_BEGIN
 
+    m_bAttach = true;
+
+    vector<CComPtr<CConfigurationSource>> configSources;
+
+    // TODO: Parse coniguration information from pvClientData.
+
+    // Mark that this is during the initialize call. This enables operations that can only be supported during initialize
+    CInitializeHolder initHolder(this);
```

Comment:
>CInitializeHolder [](start = 4, length = 17)

Should we have a different holder (ie. CInitializeAttachHolder) since there will be restrictions on SetEventMask calls, and possibly other callbacks that act differently between initialize/initializeForAttach?

Reply:
Very possible. For now, I'm going to use the same holder and add a task to create a new one that knows about what's allowed during attach. At this time, the Production Breakpoints IM works with the code as it is.

Reply:
#205